idris-repl.el

```diff
@@ -42,14 +42,28 @@
 (defvar-local idris-input-start nil
   "Marker for the start of user input for Idris.")
 
+(defun idris-repl-get-logo ()
+  "Return the path to the Idris logo if it exists, or `nil' if not."
+  (let ((logo-path (concat idris-mode-path "logo-small.png")))
+    (if (file-readable-p logo-path)
+        logo-path
+      nil)))
+
 ; TODO: insert version number / protocol version / last changed date!?
 (defun idris-repl-insert-banner ()
   "Insert Idris banner into buffer"
   (when (zerop (buffer-size))
-    (let ((welcome "Welcome to Idris REPL!"))
-      (if idris-repl-animate
-          (animate-string welcome 0 0)
-        (insert welcome)))))
+    (let ((welcome "Welcome to Idris REPL!")
+          (logo (idris-repl-get-logo)))
```

Comment:
`idris-repl-get-logo` returns `nil` if the logo file isn't found for whatever reason, so then it will just not show anything - the default case of the `cond` is missing, which means "show nothing". There's also no case for `none` for that reason, and it's advertized as a configuration value just so people know what to put in the variable.

Is it so terrible to do a cheap filesystem operation once at each REPL startup? Should it fall back to a text banner if the logo file is not found?

I can tweak the logic of it.